src/wast-parser.cc

```diff
@@ -1504,6 +1516,32 @@ Result WastParser::ParsePlainInstr(std::unique_ptr<Expr>* out_expr) {
   return Result::Ok;
 }
 
+//V: Simd Lane Immediate data type.
+template<typename V>
+Result WastParser::ParseSimdConstWithCheck(Opcode opcode, 
+                                           Type type,
+                                           Location loc,
+                                           Const* const_) {
+  WABT_TRACE(ParseSimdConstWithCheck);
+
+  const_->type = type;
+  CHECK_RESULT(ParseSimdConst(const_, Type::I32, sizeof(V)));
+
+  switch (opcode) {
+    case Opcode::I8X16ExtractLaneS: {
+      if(const_->v128_bits.v[0] >= 16)
```

Comment:
Better to do this check in the validator, so it can be shared when reading text or binary.

Reply:
Ok, I will move the check into validator.

Is it ok to omit such type of error in parsing stage?

Reply:
Yes, I think so. Any module parsed via text will have to be validated before it can be used in most cases anyway.

Reply:
Done.